typescript/web/src/components/dataset-images-list/delete-image-modal.tsx

```diff
@@ -37,6 +43,23 @@ const DELETE_IMAGE_MUTATION = gql`
   }
 `;
 
+const useRefetchAllImages = () => {
+  const client = useApolloClient();
+  const workspace = useOptionalWorkspace();
+  const workspaceSlug = workspace?.slug ?? "";
+  const { slug: datasetSlug } = useDataset();
+  return useCallback(async () => {
+    client.query<
+      GetAllImagesOfADatasetQuery,
+      GetAllImagesOfADatasetQueryVariables
+    >({
+      query: GET_ALL_IMAGES_OF_A_DATASET_QUERY,
+      variables: { slug: datasetSlug, workspaceSlug },
+      fetchPolicy: "network-only",
+    });
```

Comment:
Why don't you use `refetchQueries`?

Reply:
Because the query does not exist here.

Reply:
So why do we need to fetch it at all?

Reply:
To refresh the cache, which contains the wrong data. The query is not active but the data remains, so we need to refetch everything in order to have the correct data displayed.